yo1k/tic_tac_toe/test/core/test_game.py

```diff
@@ -22,242 +31,236 @@ def _new_state(
+                 Cell(2, 0), True),
+                ([[Mark.X, None, None], [Mark.X, None, None], [Mark.X, None, None]],
+                 Cell(1, 0), True),
+                ([[None, Mark.X, None], [None, Mark.X, None], [None, Mark.X, None]],
+                 Cell(1, 1), True),
+                ([[None, None, Mark.X], [None, None, Mark.X], [None, None, Mark.X]],
+                 Cell(1, 2), True),
+                ([[Mark.X, None, None], [None, Mark.X, None], [None, None, Mark.X]],
+                 Cell(1, 1), True),
+                ([[None, None, Mark.X], [None, Mark.X, None], [Mark.X, None, None]],
+                 Cell(1, 1), True),
+                ([[Mark.X, Mark.X, None], [Mark.X, None, None], [None, None, None]],
+                 Cell(1, 0), False),
+                ([[Mark.X, None, None], [None, None, Mark.X], [None, Mark.X, None]],
+                 Cell(0, 0), False),
+                ([[None, Mark.X, None], [None, None, Mark.X], [None, Mark.X, None]],
+                 Cell(0, 1), False),
+                ([[Mark.X, Mark.O, Mark.X], [None, None, None], [None, None, None]],
+                 Cell(0, 2), False)]
+        for args_and_expect in args_and_expect_list:
+            cells, cell, expect = args_and_expect
+            actual = Logic.win_condition(Board(cells), cell)
+            with self.subTest(expect=expect, cells=cells):
+                self.assertEqual(actual, expect)
 
-    def test_advance_win_action(self):
-        previous_game_state_board = [
-            [g.Mark.X, None, None], [g.Mark.O, g.Mark.X, g.Mark.O], [None, None, None]]
+    def test_advance__win(self):
         state = _new_state(
-            board=g.Board(previous_game_state_board),
-            step=4)
-        act_queue_px = ListActionQueue([g.Action.new_occupy(g.Cell(2, 2))])
-        g.Logic([act_queue_px, []]).advance(state)
-        next_game_state_board = [
-            [g.Mark.X, None, None], [g.Mark.O, g.Mark.X, g.Mark.O], [None, None, g.Mark.X]]
-        self.assertEqual(state.board.cells, next_game_state_board)
+                board=Board([
+                        [Mark.X, None, None], [Mark.O, Mark.X, Mark.O], [None, None, None]]),
+                step=4)
+        Logic([
+            ListActionQueue([Action.new_occupy(Cell(2, 2))]),
+            ListActionQueue([None])]).advance(state)
+        expected_board_cells = [
+            [Mark.X, None, None], [Mark.O, Mark.X, Mark.O], [None, None, Mark.X]]
+        self.assertEqual(state.board.cells, expected_board_cells)
         self.assertEqual(state.step, 4)
         self.assertEqual(state.round, 0)
-        self.assertEqual(state.phase, g.Phase.OUTROUND)
+        self.assertEqual(state.phase, Phase.OUTROUND)
         self.assertEqual(state.players[0].wins, 1)
         self.assertEqual(state.players[1].wins, 0)
 
-    def test_advance_draw_action(self):
+    def test_advance__draw(self):
         previous_game_state_board = [
```

Comment:
I wonder if you will ever actually fix all the `*state_board` variable names. You tried [twice](https://github.com/Yo1k/tic-tac-toe/pull/9#discussion_r785382499), and they are still in the code. You actually don't need this variable here at all.

Reply:
Ah, I found the change: the commit that was supposed to change only formatting actually fixes this line.